docs/blog/2018-10-26-introducing-gatsby-themes/index.md

```diff
@@ -0,0 +1,154 @@
+decisions we made for our first site. Wouldn't it be great if we could
+re-use our gatsby-config across projects? That's where starters come
+in.
+
+## Starters
+
+One way to create more sites with similar functionality faster is to
+use starters. Starters are basically whole Gatsby sites that can be
+scaffolded through the gatsby cli. This helps you start your project
+by cloning the boilerplate, installing dependencies, and clearing Git
+history. The community around gatsby has built a lot of different
+starters for various use cases including blogging, working with
+material design, and documentation.
+
+The problem with starters is that they're one-offs. Starters are
+boilerplate projects that begin to diverge immediately from upstream
+and have no easy way of updating when changes are made
+upstream. There's another approach to boilerplate that has become
+popular in recent years that fixes some problems with the boilerplate
+approach such as updating with upstream. One such project is
+create-react-app. In the gatsby world, we can improve on starters
+similarly with themes.
+
+## Themes
+
+If a single gatsby-config encodes the functionality of a whole Gatsby
+site then if we can compose the gatsby-config data structure together
+we have the base for themes. We can encode portions of our
+gatsby-config as themes and re-use them across sites.
+
+# Why Themes?
+
+Defining themes as the composition of Gatsby sites allows us to start
+solving a variety of use cases. For example, when a site gets built as
+part of a wider product offering it's often the case that one team
+will build out a set of functionality, including branding elements,
+and the other teams will mostly consume this functionality. Themes
+allow us to distribute this functionality as an npm package and allow
+the customization of various branding elements through our
+gatsby-config.
+
+# Mechanics of Theming
+
+At a base level, theming simply combines the gatsby-config of the
+theme with the gatsby-config of your site. Since it's an experimental
+feature, we use an experimental namespace to declare themes in
+gatsby-config.
+
+```js
+module.exports = {
+  __experimentalThemes: [["gatsby-theme-blog", {}]],
+}
+```
+
+Themes often need to be parameterized for various reasons, such as
+changing the base url for subsections of a site or applying branding
+variables. We can do this through the theme options if we define our
+theme's gatsby-config as a function that returns an object.
+
+```js
+module.exports = {
+  __experimentalThemes: [
+    [
+      "gatsby-theme-blog",
+      {
+        some: "value",
+      },
+    ],
+  ],
+}
+```
+
+Themes also function as plugins and any config passed into the theme
+in your gatsby-config will also be passed to your theme's gatsby-\*
+files as plugin options. This allows themes to override any settings
+inherited from the theme's own plugin declarations or apply gatsby
+lifecycle hooks such as on-create-page.
```

Comment:
```suggestion
lifecycle hooks such as [`onCreatePage`](https://www.gatsbyjs.org/docs/node-apis/#onCreatePage).
```

Reply:
shouldn't this be `(/docs/node-apis/#onCreatePage)`?

Reply:
@ChristopherBiscardi yes 👍

Reply:
Fixed, sorry.